substrate/frame/beefy/src/lib.rs

```diff
@@ -284,6 +295,78 @@ pub mod pallet {
 			GenesisBlock::<T>::put(Some(genesis_block));
 			Ok(())
 		}
+
+		/// Report voter voting on invalid fork. This method will verify the
+		/// invalid fork proof and validate the given key ownership proof
+		/// against the extracted offender. If both are valid, the offence
+		/// will be reported.
+		// TODO: fix key_owner_proofs[0].validator_count()
```

Comment:
Still needs to be fixed I assume :)

Reply:
Done in https://github.com/lederstrumpf/polkadot-sdk/commit/70785985ec9. In a valid equivocation report, the validator set count will be the same across key ownership proofs, so it's fine to read this from the first one imo. What had to be fixed though (which is what the TODO should have been formulated for), is to account for the length of the `key_owner_proofs` vec when calculating the weight.